### PR TITLE
Feat (Aave): Return v3 Debt tokens as liabilities and some fixes for #7689

### PR DESCRIPTION
**Pull request by OjusWiZard**
## Checklist

- [x] Detect Aave v3 debt tokens in balances and return them as liabilities.
- [x] Update `evm_tokens.protocol` for already existing tokens when updating Aave v3 reserve tokens.
- [x] Save the last time that task ran.

### CODE REVIEW

File: rotkehlchen/tests/unit/test_protocol_balances.py [hunk 1]
```diff
@@ -342,3 +354,83 @@ def test_gmx_balances(
...
+
+    def mock_new_balances(addresses: 'Sequence[ChecksumEvmAddress]') -> TokenBalancesType:
+        return {
+            addresses[0]: {a_eth_usdc: FVal(123), stable_debt_eth_usdc: FVal(456)},
+            addresses[1]: {stable_debt_eth_usdc: FVal(456), variable_debt_eth_usdc: FVal(789)},
+        }, {
+            a_eth_usdc: Price(FVal(0.1)),
+            stable_debt_eth_usdc: Price(FVal(10)),
+            variable_debt_eth_usdc: Price(FVal(100)),
+        }
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_protocol_balances.py:399]: can you move this to a variable `aave_prices` so you can later do

```
usd_value=amount * aave_prices[asset]
```

File: rotkehlchen/chain/aggregator.py [hunk 2]
```diff
@@ -881,6 +881,8 @@ def _update_balances_after_token_query(
                 )
                 if dsr_proxy_append is True:
                     balances[account].assets[token] += balance
+                elif token.is_liability():
+                    balances[account].liabilities[token] = balance
```
Comment (yabirgb) [inline rotkehlchen/chain/aggregator.py:885]: It is ensured that here we have only tokens in the type check

File: rotkehlchen/chain/evm/decoding/aave/v3/constants.py [hunk 3]
```diff
@@ -10,3 +11,4 @@
 BORROW: Final = b'\xb3\xd0\x84\x82\x0f\xb1\xa9\xde\xcf\xfb\x17d6\xbd\x02U\x8d\x15\xfa\xc9\xb0\xdd\xfe\xd8\xc4e\xbcsY\xd7\xdc\xe0'  # noqa: E501
 REPAY: Final = b'\xa54\xc8\xdb\xe7\x1f\x87\x1f\x9f50\xe9zt`\x1f\xea\x17\xb4&\xca\xe0.\x1cZ\xeeB\xc9lx@Q'  # noqa: E501
 BURN: Final = b'L\xf2[\xc1\xd9\x91\xc1u)\xc2R\x13\xd3\xcc\x0c\xda)^\xea\xad_\x13\xf3a\x96\x9b\x12\xeaH\x01_\x90'  # noqa: E501
+DEBT_TOKEN_SYMBOL_REGEX: Final = re.compile(r'^(stableDebt|variableDebt)\w+$')
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/aave/v3/constants.py:14]: Have you checked if in all the chains it is the same?

**Comment by yabirgb** [comment]
Also is this the last part? If so you should add the changelog entry

File: rotkehlchen/chain/evm/decoding/aave/v3/constants.py (hunk 3 above)
Comment (OjusWiZard) [reply rotkehlchen/chain/evm/decoding/aave/v3/constants.py:14]: yes, it seems. Doing this change and then running `rotkehlchen/tests/unit/test_tasks_manager.py::test_maybe_update_aave_v3_underlying_assets`, only prints the ATokens.
image
```
aEthWETH
aEthwstETH
aEthWBTC
aEthUSDC
aEthDAI
aEthLINK
aEthAAVE
aEthcbETH
aEthUSDT
aEthrETH
aEthLUSD
aEthCRV
aEthMKR
aEthSNX
aEthBAL
aEthUNI
aEthLDO
aEthENS
aEth1INCH
aEthFRAX
aEthGHO
aEthRPL
aEthsDAI
aEthSTG
aEthKNC
aEthFXS
aEthcrvUSD
aEthPYUSD
aOptDAI
aOptLINK
aOptUSDC
aOptWBTC
aOptWETH
aOptUSDT
aOptAAVE
aOptSUSD
aOptOP
aOptwstETH
aOptLUSD
aOptMAI
aOptrETH
aOptUSDCn
aPolDAI
aPolLINK
aPolUSDC
aPolWBTC
aPolWETH
aPolUSDT
aPolAAVE
aPolWMATIC
aPolCRV
aPolSUSHI
aPolGHST
aPolBAL
aPolDPI
aPolEURS
aPolJEUR
aPolAGEUR
aPolMIMATIC
aPolSTMATIC
aPolMATICX
aPolwstETH
aPolUSDCn
aArbDAI
aArbLINK
aArbUSDC
aArbWBTC
aArbWETH
aArbUSDT
aArbAAVE
aArbEURS
aArbwstETH
aArbMAI
aArbrETH
aArbLUSD
aArbUSDCn
aArbFRAX
aArbARB
aBasWETH
aBascbETH
aBasUSDbC
aBaswstETH
aBasUSDC
aGnoWETH
aGnowstETH
aGnoGNO
aGnoUSDC
aGnoWXDAI
aGnoEURe
aGnosDAI
aScrWETH
aScrUSDC
aScrwstETH
```

**Comment by OjusWiZard** [comment]
> Also is this the last part? If so you should add the changelog entry

It's not working in all the chains yet, hopefully the last thing will be the hack to enable it for all of them.

File: rotkehlchen/assets/asset.py [hunk 4]
```diff
@@ -566,6 +568,13 @@ def to_dict(self) -> dict[str, Any]:
     def get_decimals(self) -> int:
         return 18 if self.decimals is None else self.decimals
 
+    def is_liability(self) -> bool:
+        """Returns True if the token is a liability token, False if it's an asset token"""
+        return (
+            self.protocol == CPT_AAVE_V3 and
```
Comment (LefterisJP) [inline rotkehlchen/assets/asset.py:574]: what about Aave v2? Or other protocols?

File: rotkehlchen/tasks/manager.py [hunk 5]
```diff
@@ -785,6 +785,7 @@ def _maybe_update_aave_v3_underlying_assets(self) -> Optional[list[gevent.Greenl
             task_name='Update aave v3 underlying assets in globaldb',
             exception_is_error=True,
             method=update_aave_v3_underlying_assets,
+            user_db=self.database,
```
Comment (LefterisJP) [inline rotkehlchen/tasks/manager.py:788]: you don't need that. the aggregator already gives you access to the DB

File: rotkehlchen/tests/api/test_balances.py [hunk 6]
```diff
@@ -917,17 +918,19 @@ def test_blockchain_balances_refresh(rotkehlchen_api_server: 'APIServer', ethere
     )]
     vaults_patch = patch('rotkehlchen.chain.ethereum.modules.makerdao.vaults.MakerdaoVaults.get_vaults', side_effect=lambda: makerdao_vault)  # noqa: E501
 
+    a_usdc = EvmToken(identifier=A_USDC.identifier)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_balances.py:921]: just use A_USDC.resolve_to_token() isntead here and below to initialize it. And then keep what you did for using it as a variable

File: rotkehlchen/tests/unit/test_protocol_balances.py [hunk 7]
```diff
@@ -342,3 +354,85 @@ def test_gmx_balances(
...
         ),
     }
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [[
+    '0xecdED8b1c603cF21299835f1DFBE37f10F2a29Af', '0x8Cbac427B6967d5d84Ec4230081e2763AB3A8C92',
+]])
+def test_aave_v3_balances(blockchain: 'ChainsAggregator') -> None:
+    """Check that Aave v3 positions/liabilities are properly returned in the balances query"""
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_protocol_balances.py:364]:
```suggestion
    """Check that ethereum mainnet Aave v3 positions/liabilities are properly returned in the balances query"""
```

Also would need one for multiple chains. Either same or other test

File: rotkehlchen/assets/asset.py (hunk 4 above)
Comment (OjusWiZard) [reply rotkehlchen/assets/asset.py:574]: looks like Aave v1/v2 and other protocols are handled well with other logic, so I didn't add extra checks here..

**Review by LefterisJP — APPROVED** [review]
Merging this one but as discussed in discord the handling of v1/v2 liabilities by the zerion aggregator is a big question mark on whether we should leave it as it or use our normal way too. And if we do that then we would need to make sure zerion aggregator does not double count.

On the other hand doing that work now for v1 (super old) and v2 (deprecated and most have moved out of) may make no sense.

**Pull request merged**